backend/parser/listener/listener.go

```diff
@@ -0,0 +1,209 @@
+}
+func (s *searchListener[T]) ExitImplicit_and_search_expr(ctx *parser.Implicit_and_search_exprContext) {
+}
+
+func (s *searchListener[T]) EnterOr_search_expr(ctx *parser.Or_search_exprContext) {}
+func (s *searchListener[T]) ExitOr_search_expr(ctx *parser.Or_search_exprContext) {
+	rules := s.rules[len(s.rules)-2:]
+	s.rules = s.rules[:len(s.rules)-2]
+	s.rules = append(s.rules, s.sb.Or(rules...))
+}
+
+func (s *searchListener[T]) EnterKey_val_search_expr(ctx *parser.Key_val_search_exprContext) {}
+func (s *searchListener[T]) ExitKey_val_search_expr(ctx *parser.Key_val_search_exprContext)  {}
+
+func (s *searchListener[T]) EnterParen_search_expr(ctx *parser.Paren_search_exprContext) {}
+func (s *searchListener[T]) ExitParen_search_expr(ctx *parser.Paren_search_exprContext)  {}
+
+func (s *searchListener[T]) EnterSearch_key(ctx *parser.Search_keyContext) {
+	s.currentKey = ctx.GetText()
+}
+func (s *searchListener[T]) ExitSearch_key(ctx *parser.Search_keyContext) {}
+
+func (s *searchListener[T]) EnterAnd_op(ctx *parser.And_opContext) {}
+func (s *searchListener[T]) ExitAnd_op(ctx *parser.And_opContext)  {}
+
+func (s *searchListener[T]) EnterOr_op(ctx *parser.Or_opContext) {}
+func (s *searchListener[T]) ExitOr_op(ctx *parser.Or_opContext)  {}
+
+func (s *searchListener[T]) EnterNegation_op(ctx *parser.Negation_opContext) {}
+func (s *searchListener[T]) ExitNegation_op(ctx *parser.Negation_opContext)  {}
+
+func (s *searchListener[T]) EnterBin_op(ctx *parser.Bin_opContext) {
+	s.currentOp = ctx.GetText()
+}
+func (s *searchListener[T]) ExitBin_op(ctx *parser.Bin_opContext) {}
+
+func (s *searchListener[T]) EnterSearch_value(ctx *parser.Search_valueContext) {
+	value := strings.Trim(ctx.GetText(), "\"")
+	s.appendRules(value)
+}
+func (s *searchListener[T]) ExitSearch_value(ctx *parser.Search_valueContext) {}
+
+func (s *searchListener[T]) VisitTerminal(node antlr.TerminalNode)      {}
+func (s *searchListener[T]) VisitErrorNode(node antlr.ErrorNode)        {}
+func (s *searchListener[T]) EnterEveryRule(ctx antlr.ParserRuleContext) {}
+func (s *searchListener[T]) ExitEveryRule(ctx antlr.ParserRuleContext)  {}
+
+func (s *searchListener[T]) appendRules(value string) {
+	// Body column filters
+	if s.currentKey == s.tableConfig.BodyColumn {
+		if strings.Contains(value, "*") {
+			if strings.HasPrefix(value, "*") {
+				value = "%" + value[1:]
+			}
+			if strings.HasSuffix(value, "*") {
+				value = value[:len(value)-1] + "%"
+			}
+
+			s.rules = append(s.rules, s.tableConfig.BodyColumn+" ILIKE "+s.sb.Var(value))
+		} else {
+			values := strings.FieldsFunc(value, isSeparator)
+			for _, v := range values {
+				s.rules = append(s.rules, "hasTokenCaseInsensitive("+s.tableConfig.BodyColumn+", "+s.sb.Var(v)+")")
+			}
+		}
+
+		return
+	}
+
+	filterKey, ok := s.tableConfig.KeysToColumns[T(s.currentKey)]
+	if !ok {
+		filterKey = fmt.Sprintf("%s['%s']", s.attributesColumn, s.currentKey)
```

Comment:
is `currentKey` is user provided, does this mean that we are potentially injecting the raw user input into the SQL without escaping it?

Reply:
Ah, yeah, now I understand why we needed some of our existing logic for querying on custom attribute keys. Resolved in https://github.com/highlight/highlight/pull/7372/commits/55d0d8cb276d797229e15d2319e394dbdf2673bc.